pyfolio/utils.py

```
@@ -156,12 +167,24 @@ def get_returns_cached(filepath, update_func, latest_dt, **kwargs):
     if update_cache:
         returns = update_func(**kwargs)
         try:
-            mkdir(cache_dir())
+            ensure_directory(cache_dir())
+        except OSError as e:
+            warnings.warn(
+                'could not update cache: {}. {}: {}'.format(
+                    filepath, type(e).__name__, e,
+                ),
+                UserWarning,
+            )
+
+        try:
             returns.to_csv(filepath)
-        except IOError as e:
-            warnings.warn('Could not update cache {}.'
-                          'Exception: {}'.format(filepath, e),
-                          UserWarning)
+        except Exception as e:
```

I think that we should suppress any errors that come out of `to_csv`. This is non-fatal behavior and the warning is sufficient to signal that the caching has failed.